src/main/java/org/springframework/data/redis/core/types/Expiration.java

```diff
@@ -219,6 +220,25 @@ public boolean isUnixTimestamp() {
 		return false;
 	}
 
+	@Override
+	public boolean equals(Object obj) {
+
+		if (this == obj) {
+			return true;
+		}
+
+		if (!(obj instanceof Expiration that)) {
+			return false;
+		}
+
+		return this.getTimeUnit().toMillis(getExpirationTime()) == that.getTimeUnit().toMillis(that.getExpirationTime());
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(getExpirationTime(), getTimeUnit());
```

Comment:
We use Spring's `ObjectUtils.nullSafeHashCode(…)` to compute the hash code in other places.

Reply:
Right! Old habit.

Reply:
Spring Framework's `ObjectUtils.nullSafeHashCode(..)` is a bit more verbose and sloppy, but no matter.